accepted/install-locations.md

```diff
@@ -0,0 +1,241 @@
+* Enable multi-level on Linux/macOS in its current form - making it consistent with Windows.
+
+Proposal:
+**Leave it as-is on Linux/macOS.**  
+Pros:
+* Most important scenarios (as described above) will work just fine if we add the global locations into `hostfxr` search. The scenarios where multi-level lookup helps don't work on Linux/macOS today and we haven't got much feedback to enable it.
+* Avoids extending the controversial multi-level lookup to Linux/macOS. Once we figure out the replacement story, that should be implemented on Linux/macOS, but there won't be any backward compat burden.
+
+Cons:
+* Inconsistent behavior between Windows and Linux/macOS
+
+## Discussed questions
+
+### Make muxer less special
+Proposal is to leave muxer as is - don't use `DOTNET_ROOT` in it.
+
+Leaving the below as interesting discussion...
+As mentioned above, the muxer has certain specific behavior:
+* It only looks next to itself when searching for `hostfxr`.
+* It ignores `DOTNET_ROOT` environment variable.
+
+#### Framework and SDK search
+I think the muxer should use `DOTNET_ROOT` to resolve frameworks and SDKs just like all the other hosts (the actual implementation mechanism aside). The current discrepancy is simply unexpected and makes certain scenarios unnecessarily complex (for example VS can't really use private installs because of this, since it always uses the muxer from path).
+
+In itself this change has no real downside. It has a small potential for being breaking: If an application uses the muxer and also has the `DOTNET_ROOT` set, previously the environment variable would be ignored and thus the app would only find frameworks/SDKs in the global locations. After the change to muxer, the muxer would suddenly also find frameworks/SDKs in the custom location.
+* For frameworks this is very little risk. The only observable change would be to an app which for example requires framework version `2.0.0`, but the machine only has `2.1.0` in the global location. If the custom location in `DOTNET_ROOT` would have for example `2.0.1` then after the change the app would start using the private `2.0.1` instead. In theory this is for the better as that framework better fulfills the app requirements.
+* For SDKs this is of a slightly higher risk. If the custom location in `DOTNET_ROOT` contains SDK with higher version than that available in global location, the default selection would be the higher version from the custom location. If the app uses the SDK to create new applications this might also change the TFM of the created apps. SDKs should be backward compatible though, so the newer SDK should work fine on older apps anyway.
+
+#### `hostfxr` search
+Where the muxer should search for `hostfxr` is a more complicated issue.
+As of 2.2 only the location right next to the muxer is considered. This effectively means that `hostfxr` should always be equal or newer version than the muxer itself.
```

Comment:
There are security implications to why this was done this way.

`dotnet.exe` is typically installed to a "trusted" location - ProgramFiles. If you can "trick" this trusted app to go loading up a "framework" that is installed into an "untrusted" location, then you potentially have a security issue. The scenario is: I download a malicious framework into the `Downloads` folder, and then set `DOTNET_ROOT` environment variable to `Downloads`, now when someone invokes `dotnet.exe`, it loads my malicious framework.